src/settings.ts

```diff
@@ -138,10 +138,10 @@ export default class Settings {
     uri: string,
     stylelint: typeof globalStylelint
   ): typeof globalStylelint["lint"] {
-    return (...args) => {
+    return async (...args) => {
       if (!this.failedDocuments.has(uri)) {
         try {
-          return stylelint.lint(...args)
+          return await stylelint.lint(...args)
```

Comment:
I decided to just stick an `await` here and not rewrite the function to use a `.catch`, since this solution reuses the "default value" returned from the function below in the case of an error